### PR TITLE
C# highlighting improvements

### PR DESCRIPTION
**Pull request by Philipp-M**
Adds more queries for better highlighting:

before:
helix-csharp-highlight-before

after:
helix-csharp-highlight-improvement

### CODE REVIEW

File: runtime/queries/c-sharp/highlights.scm [hunk 1]
```diff
@@ -236,3 +311,14 @@
...
+;; Rest
+(member_access_expression) @variable
+(element_access_expression (identifier) @variable)
+(argument (identifier) @variable)
+(for_statement (identifier) @variable)
+(for_each_statement (identifier) @variable)
+(expression_statement (identifier) @variable)
+(member_access_expression expression: (identifier) @variable)
+(member_access_expression name: (identifier) @variable)
+(conditional_access_expression [(identifier) (qualified_name)] @variable)
```
Comment (the-mikedavis) [inline runtime/queries/c-sharp/highlights.scm:324]: would you mind adding an extra newline to the end here?

**Review by the-mikedavis — COMMENTED** [review]
I haven't dug into this fully yet but it's looking like a great improvement! There are some highlights I've seen that I think would be good to include here as well if you like them:

```scm
; this list already exists and we should be able to just add "static" in
[
  ; ..
  "static"
  ; ..
] @keyword
```

highlights a block like so:

```cs
using Status = Google.Rpc.Status;
using static EventStore.Client.Streams.BatchAppendReq.Types;
using static EventStore.Client.Streams.BatchAppendReq.Types.Options;
using OperationResult = EventStore.Core.Messages.OperationResult;
```

```scm
((identifier) @comment.unused
 (#eq? @comment.unused "_"))
```

we've used in other languages that support pattern matching to mark it as an unused pattern.

For example in an expression like so:

```cs
Task.Delay(timeout, cancellationToken).ContinueWith(_ => onTimeout(), cancellationToken);
```

**Comment by Philipp-M** [comment]
Thanks for the input, I've adapted your changes

File: runtime/queries/c-sharp/highlights.scm [hunk 2]
```diff
@@ -19,6 +51,35 @@
...
 
+;; Generic Types
+(type_of_expression
+  (generic_name
+    (identifier) @type))
+
+; (type_argument_list) @type
+;   (generic_name
+;     (identifier) @type))
+
```
Comment (the-mikedavis) [inline runtime/queries/c-sharp/highlights.scm:62]: looks like this got moved down to L299, should be safe to remove this comment

**Review by the-mikedavis — COMMENTED** [review]
one more highlight I found looking at some example code:

```scm
[
  ; .. L148
  ">"
  ">="
] @operator
```

With that I think this will be ready to merge 👍

**Comment by Philipp-M** [comment]
I missed that there were operators that aren't covered yet, I checked this: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/#operator-precedence

And I think I covered now every operator in one way or the other (something like "?." was not possible because it's already covered by "." and "?" I guess?)

**Comment by the-mikedavis** [comment]
It's a bit confusing what can be written in strings with the query DSL: you can only write literal tokens that show up also as strings in the grammar (for example, [these tokens](https://github.com/tree-sitter/tree-sitter-c-sharp/blob/352a4630c81a7a5cbd3bc67327743bd8d38f2dd2/grammar.js#L1139)). I don't see `?.` in `grammar.js` - there must be some rule that parses the `?` and `.` as separate tokens.

**Review by the-mikedavis — APPROVED** [review]
Great work! This is a big improvement 💪

**Pull request merged**